### PR TITLE
Camel 9733 Add support for JobName specified in the message.

### PR DESCRIPTION
**Pull request by jlpedrosa**
Hi. 
This is a fresh startup of: https://github.com/apache/camel/pull/968.

Thanks gius

### CODE REVIEW

File: components/camel-spring-batch/src/main/java/org/apache/camel/component/spring/batch/SpringBatchEndpoint.java [hunk 1]
```diff
@@ -37,15 +38,21 @@
 @UriEndpoint(scheme = "spring-batch", title = "Spring Batch", syntax = "spring-batch:jobName", producerOnly = true, label = "spring,batch,scheduling")
 public class SpringBatchEndpoint extends DefaultEndpoint {
 
-    @UriPath @Metadata(required = "true")
+
```
Comment (trohovsky) [inline components/camel-spring-batch/src/main/java/org/apache/camel/component/spring/batch/SpringBatchEndpoint.java]: redundant blank line

File: components/camel-spring-batch/src/main/java/org/apache/camel/component/spring/batch/SpringBatchEndpoint.java [hunk 2]
```diff
@@ -144,4 +151,19 @@ public JobLauncher getJobLauncher() {
     public void setJobLauncher(JobLauncher jobLauncher) {
         this.jobLauncher = jobLauncher;
     }
+
```
Comment (trohovsky) [inline components/camel-spring-batch/src/main/java/org/apache/camel/component/spring/batch/SpringBatchEndpoint.java]: redundant blank line

File: components/camel-spring-batch/src/main/java/org/apache/camel/component/spring/batch/SpringBatchEndpoint.java [hunk 3]
```diff
@@ -144,4 +151,19 @@ public JobLauncher getJobLauncher() {
...
+
+    /**
+     * Explicitly defines if the jobName shouls be taken from the headers instead of the URI.
+     */
+    public void setJobFromHeader(Boolean jobFromHeader) {
+        this.jobFromHeader = jobFromHeader;
+    }
+
```
Comment (trohovsky) [inline components/camel-spring-batch/src/main/java/org/apache/camel/component/spring/batch/SpringBatchEndpoint.java]: redundant blank line

File: components/camel-spring-batch/src/main/java/org/apache/camel/component/spring/batch/SpringBatchEndpoint.java [hunk 4]
```diff
@@ -144,4 +151,19 @@ public JobLauncher getJobLauncher() {
...
+
+    public Boolean getJobFromHeader() {
+        return jobFromHeader;
+    }
+
+
```
Comment (trohovsky) [inline components/camel-spring-batch/src/main/java/org/apache/camel/component/spring/batch/SpringBatchEndpoint.java]: redundant blank line

File: components/camel-spring-batch/src/test/java/org/apache/camel/component/spring/batch/SpringBatchEndpointTest.java [hunk 5]
```diff
@@ -55,33 +57,90 @@
...
+
     // Camel fixtures
 
     @EndpointInject(uri = "mock:test")
     MockEndpoint mockEndpoint;
 
+    @EndpointInject(uri = "mock:error")
+    MockEndpoint errorEndpoint;
+
+
```
Comment (trohovsky) [inline components/camel-spring-batch/src/test/java/org/apache/camel/component/spring/batch/SpringBatchEndpointTest.java]: redundant blank line

File: components/camel-spring-batch/src/test/java/org/apache/camel/component/spring/batch/SpringBatchEndpointTest.java [hunk 6]
```diff
@@ -55,33 +57,90 @@
...
                 from("direct:start").to("spring-batch:mockJob").to("mock:test");
+                from("direct:dynamic").
+                        to("spring-batch:fake?jobFromHeader=true").
+                        errorHandler(deadLetterChannel("mock:error")).
+                        to("mock:test");
             }
         };
     }
 
+
```
Comment (trohovsky) [inline components/camel-spring-batch/src/test/java/org/apache/camel/component/spring/batch/SpringBatchEndpointTest.java]: redundant blank line

File: components/camel-spring-batch/src/test/java/org/apache/camel/component/spring/batch/SpringBatchEndpointTest.java [hunk 7]
```diff
@@ -55,33 +57,90 @@
...
 
     // Tests
 
     @Test
+    public void dynamicJobFailsIfHeaderNotPressent() throws Exception {
+
+        mockEndpoint.expectedMessageCount(0);
+        errorEndpoint.expectedMessageCount(1);
+
+        //dynamic job should fail as header is not present and the job is dynamic
```
Comment (trohovsky) [inline components/camel-spring-batch/src/test/java/org/apache/camel/component/spring/batch/SpringBatchEndpointTest.java]: redundant comment, the test name is enough descriptive (btw. missing space after //)

File: components/camel-spring-batch/src/test/java/org/apache/camel/component/spring/batch/SpringBatchEndpointTest.java [hunk 8]
```diff
@@ -55,33 +57,90 @@
...
+        mockEndpoint.assertIsSatisfied();
+    }
+
+    @Test
+    public void dynamicJobFailsIfHeaderWithInvalidJobName() throws Exception {
+
+        mockEndpoint.expectedMessageCount(0);
+        errorEndpoint.expectedMessageCount(1);
+
+        //dynamic job should fail as header is present but the job does not exists
```
Comment (trohovsky) [inline components/camel-spring-batch/src/test/java/org/apache/camel/component/spring/batch/SpringBatchEndpointTest.java]: redundant comment, the test name is enough descriptive

File: components/camel-spring-batch/src/test/java/org/apache/camel/component/spring/batch/SpringBatchEndpointTest.java [hunk 9]
```diff
@@ -55,33 +57,90 @@
...
+        mockEndpoint.assertIsSatisfied();
+    }
+
+    @Test
+    public void dynamicJobWorksIfHeaderPressentWithValidJob() throws Exception {
+
+        mockEndpoint.expectedMessageCount(1);
+        errorEndpoint.expectedMessageCount(0);
+
+        //dynamic job work if header is present and the job exists
```
Comment (trohovsky) [inline components/camel-spring-batch/src/test/java/org/apache/camel/component/spring/batch/SpringBatchEndpointTest.java]: redundant comment, the test name is enough descriptive

**Comment by trohovsky** [comment]
Seems good now ;-), only some formatting issues

**Comment by jlpedrosa** [comment]
Pushed the changes with the lines removed.

@trohovsky  if blank lines is important, then pelase add it to the rules so we don't have to go back and forward with 4 comments about blank lines, it's pointless.  Also I've removed the comment lines that you find redundant, but honestly I don't see the point in wasting time over a commentary line.

**Comment by davsclaus** [comment]
Thanks for the PR it has been merged. Do you mind closing this?

**Pull request closed without merging**